### PR TITLE
Bug 1849576 - Navigation toolbar TestRail matching

### PR DESCRIPTION
**Pull request by AndiAJ**
Summary:

1. TestRail matching:
These changes are part of our data alignment between testRail and our automated UI tests.
For more information please see this [document](https://docs.google.com/document/d/1erM2u9yMhOsaUV4zVFUv7czeJLpL7JaClw5fa3fkvY8/edit?usp=sharing)

2. Removed duplicate UI tests from the Compose* test classes that don't actually interact with the new composable tabs tray

3. While doing the matching, 2 UI tests were completely removed.
These 2  UI tests were running on different UI test related tasks.

@AaronMT , please let me know if we want to include other UI tests for these tasks.
- `visitURLTest` - was removed from[ flank-arm-beta.yml](https://github.com/mozilla-mobile/firefox-android/pull/3358/files#diff-9ee63d8dfcc3774cade747e584faa1f0f5a84c7e4b8d8711cc7ef60ce495b6dcL23), [flank-arm-legacy-api-tests.yml](https://github.com/mozilla-mobile/firefox-android/pull/3358/files#diff-768d1abda13f4cf200181465cc966b5a2e56aeadf9a3d148d436a9946cbbd307L26) and [flank-arm-start-test.yml](https://github.com/mozilla-mobile/firefox-android/pull/3358/files#diff-3fb27f85d6f6e96c3540c43c9d44621c1030ef8fffe782255412a533d7099b7cL23)
- `verifyBasicNavigationToolbarFunctionality` - was removed from [flank-arm-legacy-api-tests.yml](https://github.com/mozilla-mobile/firefox-android/pull/3358/files#diff-768d1abda13f4cf200181465cc966b5a2e56aeadf9a3d148d436a9946cbbd307L34)

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1849576

### CODE REVIEW

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/NavigationToolbarTest.kt [hunk 1]
```diff
@@ -142,7 +147,7 @@ class NavigationToolbarTest {
     // Test running on beta/release builds in CI:
     // caution when making changes to it, so they don't block the builds
     @Test
-    fun visitURLTest() {
+    fun basicToolbarInteractionTest() {
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/NavigationToolbarTest.kt:150]: I would delete this test entirely, we have this covered in a lot of other tests.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/NavigationToolbarTest.kt [hunk 2]
```diff
@@ -258,4 +263,102 @@ class NavigationToolbarTest {
...
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(genericURL.url) {
+            mDevice.waitForIdle()
+        }.goToHomescreen {
+            verifyHomeScreen()
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openCustomizeSubMenu {
+            clickTopToolbarToggle()
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/NavigationToolbarTest.kt:291]: I would remove the top toolbar check in these tests, as I don't think it is too relevant. We have other tests for the top toolbar position.

**Review by sv-ohorvath — APPROVED** [review]

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/NavigationToolbarTest.kt (hunk 2 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/NavigationToolbarTest.kt]: As discussed, I've removed both the step from the manual and removed the corresponding automated test.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/NavigationToolbarTest.kt (hunk 1 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/NavigationToolbarTest.kt]: As discussed, I've removed both the manual and automated test.

**Review by AaronMT — APPROVED** [review]

**Review by isabelrios — APPROVED** [review]

**Comment by isabelrios** [comment]
Thanks @AndiAJ ! let's land this once the conflicts are fixed.

**Pull request merged**